src/test/java/org/jenkinsci/plugins/workflow/EnvWorkflowTest.java

```diff
@@ -47,13 +47,14 @@
     @Test public void isNodeNameAvailable() throws Exception {
         r.createSlave("node-test", "unix fast", null);
         WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "workflow-test");
+        final String builtInNodeName = r.jenkins.getSelfLabel().getName();
 
         p.setDefinition(new CpsFlowDefinition(
             "node('master') {\n" +
             "  echo \"My name on master is ${env.NODE_NAME} using labels ${env.NODE_LABELS}\"\n" +
             "}\n",
             true));
-        r.assertLogContains("My name on master is master using labels master", r.assertBuildStatusSuccess(p.scheduleBuild2(0)));
+        r.assertLogContains("My name on master is " + builtInNodeName + " using labels master", r.assertBuildStatusSuccess(p.scheduleBuild2(0)));
```

Comment:
So was `Jenkins.getAssignedLabels` not updated?

Reply:
That's https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/167

Reply:
Sorry, I am lost. Is there a reason why there are two PRs adapting to the same upstream change and editing the same line of code in different ways, meaning whoever merges them will be faced with a merge conflict on the second one?

Reply:
#167 adapts the test to upstream changes to make PCT work and can basically be applied blindly. This PR changes behavior in quite significant ways. This makes them independently applicable (admittedly conflicting in an easy to resolve manner).

Reply:
it looks like this PR is also needed to make sure that the PCT does not run into the same timeout issues for 2.307+

Reply:
```suggestion
        r.assertLogContains("My name on master is " + builtInNodeName + " using labels " + builtInNodeLabel, r.assertBuildStatusSuccess(p.scheduleBuild2(0)));
```

?

Reply:
Looks like a bad merge between the PRs since both change different parts of this line. Checking locally...